Style parsing and sprite loading errors

The first time an `mbgl::Map` is created after launch, the following error appears in the console and no map appears:

```
2016-01-14 17:29:48.969 Mapbox GL[49424:5375589] [ERROR] {Map}[ParseStyle]: Error parsing style JSON at 0: The document is empty.
```

Any subsequent time an `mbgl::Map` is created, the map does load, but with the following console message:

```
2016-01-14 17:35:50.019 Mapbox GL[50015:5381602] [ERROR] {Map}[Style]: Failed to load sprite: CGImageSourceCreateImageAtIndex failed
```

No sprites load: all annotations are invisible, as are any POIs in the style that have icons. This regression was introduced sometime within the past several hours. Reproduces in both iosapp and osxapp.

/cc @kkaefer @jfirebaugh

--- a/platform/android/src/http_request_android.cpp
+++ b/platform/android/src/http_request_android.cpp
@@ -209,6 +209,8 @@ void HTTPAndroidRequest::onResponse(int code, std::string message, std::string e
         response->notModified = true;
 
         if (existingResponse) {
+            response->data = existingResponse->data;
+
             if (response->expires == Seconds::zero()) {
                 response->expires = existingResponse->expires;
             }

--- a/platform/darwin/http_request_nsurl.mm
+++ b/platform/darwin/http_request_nsurl.mm
@@ -229,6 +229,8 @@ void HTTPNSURLRequest::handleResult(NSData *data, NSURLResponse *res, NSError *e
             response->notModified = true;
 
             if (existingResponse) {
+                response->data = existingResponse->data;
+
                 if (response->expires == Seconds::zero()) {
                     response->expires = existingResponse->expires;
                 }

--- a/platform/default/http_request_curl.cpp
+++ b/platform/default/http_request_curl.cpp
@@ -532,6 +532,8 @@ void HTTPCURLRequest::handleResult(CURLcode code) {
             response->notModified = true;
 
             if (existingResponse) {
+                response->data = existingResponse->data;
+
                 if (response->expires == Seconds::zero()) {
                     response->expires = existingResponse->expires;
                 }

--- a/test/storage/cache_revalidate.cpp
+++ b/test/storage/cache_revalidate.cpp
@@ -49,6 +49,8 @@ TEST_F(Storage, CacheRevalidateSame) {
             EXPECT_EQ(nullptr, res2.error);
             EXPECT_EQ(false, res2.stale);
             EXPECT_TRUE(res2.notModified);
+            ASSERT_TRUE(res2.data.get());
+            EXPECT_EQ("Response", *res2.data);
             EXPECT_LT(Seconds::zero(), res2.expires);
             EXPECT_EQ(Seconds::zero(), res2.modified);
             // We're not sending the ETag in the 304 reply, but it should still be there.
@@ -107,6 +109,8 @@ TEST_F(Storage, CacheRevalidateModified) {
             EXPECT_EQ(nullptr, res2.error);
             EXPECT_EQ(false, res2.stale);
             EXPECT_TRUE(res2.notModified);
+            ASSERT_TRUE(res2.data.get());
+            EXPECT_EQ("Response", *res2.data);
             EXPECT_LT(Seconds::zero(), res2.expires);
             EXPECT_EQ(1420070400, res2.modified.count());
             EXPECT_EQ("", res2.etag);

